### PR TITLE
qa/tasks: watchdog should terminate thrasher

### PR DESCRIPTION
**Pull request by NitzanMordhai**
If a thrasher exception occurs, the do_dump_ops thread will continue looping until the Teuthology timeout is reached.
The watchdog should terminate the thrasher to free up resources.

Fixes: https://tracker.ceph.com/issues/66698

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by rzarzynski — APPROVED** [review]

File: qa/tasks/daemonwatchdog.py [hunk 1]
```diff
@@ -117,6 +117,7 @@ def watch(self):
             for thrasher in self.thrashers:
                 if thrasher.exception is not None:
                     self.log("{name} failed".format(name=thrasher.name))
+                    thrasher.do_join()
```
[the remark this answers is not in the document]
Comment (NitzanMordhai) [reply qa/tasks/daemonwatchdog.py]: @batrick thanks! i added those changes and also fixed the others thrashers

**Review by batrick — APPROVED** [review]

**Comment by batrick** [comment]
jenkins test make check

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
This PR is under test in [https://tracker.ceph.com/issues/66822](https://tracker.ceph.com/issues/66822).

**Comment by batrick** [comment]
@NitzanMordhai @ljflores @rzarzynski `fs` QA run looked fine. I suspect you want to run this through `rados`?

**Comment by NitzanMordhai** [comment]
Rados approved: https://tracker.ceph.com/projects/rados/wiki/MAIN#httpstrackercephcomissues67266

**Comment by yuriw** [comment]
@NitzanMordhai pls rebase and merge
ref: https://tracker.ceph.com/issues/67266

**Comment by ljflores** [comment]
@NitzanMordhai can you resolve the conflicts?

**Comment by NitzanMordhai** [comment]
> @NitzanMordhai can you resolve the conflicts?

done

**Comment by vshankar** [comment]
Sorry to comment on a merged PR, but this change seems to be causing failures like: https://pulpito.ceph.com/vshankar-2024-08-07_04:36:45-fs-wip-vshankar-testing-20240806.162829-debug-testing-default-smithi/7841593/

@NitzanMordhai

**Comment by NitzanMordhai** [comment]
@vshankar thanks, missed that one, i created https://tracker.ceph.com/issues/67496 and working on it

**Pull request merged**